x-pack/plugins/observability_solution/apm_data_access/server/services/get_host_services/index.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { rangeQuery, termQuery } from '@kbn/observability-plugin/server';
+import {
+  AGENT_NAME,
+  HOST_HOSTNAME,
+  HOST_NAME,
+  METRICSET_NAME,
+  SERVICE_NAME,
+} from '@kbn/apm-types/es_fields';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  TimeRangeMetadata,
+  getBucketSize,
+  getPreferredBucketSizeAndDataSource,
+} from '../../../common';
+import { ApmDocumentType } from '../../../common/document_type';
+import type { ApmDataAccessServicesParams } from '../get_services';
+
+const MAX_SIZE = 1000;
+
+export interface HostServicesRequest {
+  filters: Record<string, string>;
+  start: number;
+  end: number;
+  size?: number;
+  documentSources: TimeRangeMetadata['sources'];
+}
+
+const suitableTypes = [ApmDocumentType.TransactionMetric, ApmDocumentType.ErrorEvent];
+
+export function createGetHostServices({ apmEventClient }: ApmDataAccessServicesParams) {
+  return async ({ start, end, size = MAX_SIZE, filters, documentSources }: HostServicesRequest) => {
+    const sourcesToUse = getPreferredBucketSizeAndDataSource({
+      sources: documentSources.filter((s) => suitableTypes.includes(s.documentType)),
+      bucketSizeInSeconds: getBucketSize({ start, end, numBuckets: 50 }).bucketSize,
+    });
+
+    const commonFiltersList: QueryDslQueryContainer[] = [
+      ...rangeQuery(start, end),
+      {
+        exists: {
+          field: SERVICE_NAME,
+        },
+      },
+    ];
+
+    if (filters[HOST_NAME]) {
+      commonFiltersList.push({
+        bool: {
+          should: [
+            ...termQuery(HOST_NAME, filters[HOST_NAME]),
+            ...termQuery(HOST_HOSTNAME, filters[HOST_HOSTNAME]),
+          ],
+          minimum_should_match: 1,
+        },
+      });
+    }
+    // get services from transaction metrics
+    const metricsQuery = await apmEventClient.search('get_apm_host_services_from_metrics', {
+      apm: {
+        sources: [
+          {
+            documentType: sourcesToUse.source.documentType,
```

Comment:
Should we target here only `ApmDocumentType.TransactionMetric`? Previously this query wasn't querying `ApmDocumentType.ErrorEvent`